src/test/java/org/helpberkeley/memberdata/MainTest.java

```diff
@@ -851,6 +851,27 @@ public void updateMemberDataTooManyMembers() {
         assertThat(thrown).hasMessageContaining(MessageFormat.format(WorkflowExporter.TOO_MANY_MEMBERS_ERROR, 3));
     }
 
+    @Test
+    public void updateMemberDataHeaderMismatch() {
+        String deliveries = readResourceFile("update-member-data-multiple-updates.csv");
+        WorkflowParser parser = WorkflowParser.create(Collections.emptyMap(), deliveries);
+        ApiClient apiSim = createApiSimulator();
+        List<User> userList = new Loader(apiSim).load();
+        Map<String, User> users = new Tables(userList).mapByUserName();
+        String json = apiSim.runQuery(Constants.QUERY_GET_DELIVERY_DETAILS);
+        ApiQueryResult apiQueryResult = HBParser.parseQueryResult(json);
+        Map<String, DetailsPost> deliveryDetails = HBParser.deliveryDetails(apiQueryResult);
+        WorkflowExporter exporter = new WorkflowExporter(parser);
+        String wrongHeader = "Consumer,Driver,Name,User Fame,Phone #,Phone2 #,City,Neighborhood,Address,Condo," +
+                "Details,Restaurants,std meals,alt meals,type meal,std grocery,alt grocery,type grocery";
+        String rightHeader = "Consumer,Driver,Name,User Name,Phone #,Phone2 #,Neighborhood,City,Address,Condo," +
+                "Details,Restaurants,std meals,alt meals,type meal,std grocery,alt grocery,type grocery";
+        exporter.changeIncomingHeader(wrongHeader);
```

Comment:
This can't be reached in the normal path, because we already would have failed when constructing the WorkflowParser, but here we are modifying the header after it has already been constructed.   To see an example of this auditing see the WorkflowHBParserBaseTest.missingHeaderColumnTest() that I just added to the master branch.

What I thought we wanted to see here was what the case where there were some columns of data without header columns.